### PR TITLE
Updated SignupActivity to new Design pattern plus refractoring. 

### PR DESCRIPTION
**Pull request by mayank408**
Fixes issue #825

Updated SignupActivity to new Design pattern plus refractoring

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt [hunk 1]
```diff
@@ -128,4 +128,4 @@ class LoginPresenter(loginActivity: LoginActivity): ILoginPresenter, ILoginModel
     override fun onDetach() {
         loginView = null
     }
-}
+}
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt]: It is a good practice to have a newline at the end of the file. Not compulsory but a suggestion.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -122,6 +121,7 @@
     <string name="first_time">FIRST_TIME</string>
     <string name="ok">OK</string>
     <string name="invalid_email">Invalid Email</string>
+    <string name="email_invalid_title">Incorrect Email</string>
```
Comment (chiragw15) [inline app/src/main/res/values/strings.xml]: Suggestion: Avoid these kinds of changes (Shifting of one line from one place to another). They cause conflicts.

File: app/src/main/java/org/fossasia/susi/ai/signup/contract/ISignUpView.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package org.fossasia.susi.ai.signup.contract
+
+/**
+ *  * The interface for SignUp view
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/signup/contract/ISignUpView.kt]: Fix this maybe?

File: app/src/main/java/org/fossasia/susi/ai/signup/contract/ISignUpPresenter.kt [hunk 4]
```diff
@@ -1,8 +1,11 @@
-package org.fossasia.susi.ai.signup
+package org.fossasia.susi.ai.signup.contract
 
 /**
+ * * The interface for SignUp Presenter
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/signup/contract/ISignUpPresenter.kt]: Optional : Fix this

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt [hunk 5]
```diff
@@ -1,82 +1,119 @@
...
     var signUpView: ISignUpView? = null
-    var signUpInteractor: SignUpInteractor? = null
+    var signUpModel: SignUpModel? = null
+    var utilModel: UtilModel = UtilModel(signUpActivity)
+    lateinit var email: String
 
     override fun onAttach(signUpView: ISignUpView) {
         this.signUpView = signUpView
-        this.signUpInteractor = SignUpInteractor()
+        this.signUpModel = SignUpModel()
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: Move this from onAttach to constructor.

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt [hunk 6]
```diff
@@ -1,82 +1,119 @@
...
+ *
  * Created by mayanktripathi on 05/07/17.
  */
 
-class SignUpPresenter: ISignUpPresenter, ISignUpInteractor.OnLoginFinishedListener {
+class SignUpPresenter(signUpActivity: SignUpActivity) : ISignUpPresenter, ISignUpModel.OnSignUpFinishedListener {
 
     var signUpView: ISignUpView? = null
-    var signUpInteractor: SignUpInteractor? = null
+    var signUpModel: SignUpModel? = null
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: Avoid nullable variables. Use constructor and initialize it

File: app/src/main/java/org/fossasia/susi/ai/helper/Constant.java [hunk 7]
```diff
@@ -49,9 +49,10 @@
 
     public static final String ANONYMOUS_LOGGED_IN = "anonymous_logged_in";
     public static final String FIRST_TIME = "first_time";
-    public static final CharSequence THEME_KEY = "Theme_key";
 
     public static final String EMAIL = "email";
     public static final String PASSWORD = "password";
+    public static final String CONFIRM_PASSWORD = "confirm_password";
+    public static final CharSequence THEME_KEY = "Theme_key";
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/helper/Constant.java]: Any particular reason to have this as CharSequence?

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISignUpModel.kt [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package org.fossasia.susi.ai.data.contract
+
+import org.fossasia.susi.ai.rest.responses.susi.SignUpResponse
+import retrofit2.Response
+
+/**
+ * Created by mayanktripathi on 10/07/17.
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/ISignUpModel.kt]: Add info here too.

**Comment by mayank408** [comment]
@mariobehling Pls merge this

**Comment by mariobehling** [comment]
Screenshots?

**Comment by mayank408** [comment]
@mariobehling This is change in MVP design of architecture of the app and not related to the UI design of the app.

**Pull request merged**